src/utils/api.js

```diff
@@ -0,0 +1,77 @@
+const BASE_URL =
+  'https://api.news-explorer-gregory.students.nomoredomainssbs.ru';
+function checkResponce(res) {
+  if (res.ok) {
+    return res.json();
+  }
+  return Promise.reject(`Error: ${res.status}`);
+}
+function generateHeaders() {
+  const jwt = localStorage.getItem('jwt');
+  return {
+    Accept: 'application/json',
+    Authorization: `Bearer ${jwt}`,
+    'Content-Type': 'application/json',
+  };
+}
+export const register = ({ email, password, username }) => {
+  return fetch(`${BASE_URL}/signup`, {
+    method: 'POST',
+    headers: {
+      Accept: 'application/json',
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({ email, password, name: username }),
+  });
+};
+
+export const login = ({ email, password }) => {
+  return fetch(`${BASE_URL}/signin`, {
+    method: 'POST',
+    headers: {
+      Accept: 'application/json',
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({ email, password }),
+  });
+};
+
+export const checkToken = () => {
+  return fetch(`${BASE_URL}/users/me`, {
+    method: 'GET',
+    headers: generateHeaders(),
+  }).then(checkResponce);
```

Comment:
**Might be better**

You can make a special method for fetching and checking responses **not to duplicate** it in every request:

```js

  function request(url, options) {
    return fetch(url, options).then(checkResponse)
  }

```

And now you can safely replace `fetch` with `request`. And you don’t need to change anything else.